addon/decorators/create-changeset.js

```diff
@@ -0,0 +1,23 @@
+import { get, set } from '@ember/object';
+import { copy } from '@ember/object/internals';
+import { isArray } from '@ember/array';
+import Changeset from 'ember-changeset';
+import lookupValidator from 'ember-changeset-validations';
+
+export default function createChangeset(target, validations, cloneAttrs) {
+    const changeset = validations ?
+        new Changeset(target, lookupValidator(validations), validations) :
+        new Changeset(target);
+
+    //create a copy of attributes whose values are passed by reference
+    //and set the copy on the changeset so that changes do not affect the target
+    if(isArray(cloneAttrs)) {
+        cloneAttrs.forEach((attr) => {
+            let value = get(target, attr);
+            let clonedValue = isArray(value) ? value.toArray() : copy(value);
+            set(changeset, attr, clonedValue);
+        });
+    }
+
+    return changeset;
+}
```

Comment:
we should just make this a standard utility function under `addon/utilities/create-changeset.js`, and then we can also get rid of the create-changeset mixin. This function doesn't need any access to use or override class instance methods/properties, so theres no reason for it to pollute the class namespace.

Then in the changeset-route decorator, we can simply import it and use it like you would any other function.

Reply:
@billdami Im not sure we should get rid of the mixin. That would mean that we have to do a new major release and I was trying to prevent that. Keeping that mixin means we keep backwards compatibility